src/curl_async.c

```diff
@@ -523,7 +524,7 @@ static int process_download(struct swupd_curl_parallel_handle *h, struct multi_c
 	return poll_fewer_than(h, h->max_xfer, h->max_xfer / 2);
 
 out_bad:
-	return -1;
+	return -curl_ret;
```

Comment:
I don't understant the idea here. Why returning a negative curl_ret? and when you set it to -1 it will return 1. Is this expected?

Reply:
ups, my bad, I meant to return -1 here.

I decided to return curl_ret so we could have more information about the error type instead of just returning a -1 for every failure, that way we have the possibility of responding differently depending on the failure